src/wallet/wallet.cpp

```diff
@@ -1929,16 +1929,7 @@ bool CWallet::FundTransaction(CMutableTransaction& tx, CAmount &nFeeRet, int& nC
     // Add new txins (keeping original txin scriptSig/order)
     BOOST_FOREACH(const CTxIn& txin, wtx.vin)
     {
-        bool found = false;
-        BOOST_FOREACH(const CTxIn& origTxIn, tx.vin)
-        {
-            if (txin.prevout.hash == origTxIn.prevout.hash && txin.prevout.n == origTxIn.prevout.n)
-            {
-                found = true;
-                break;
-            }
-        }
-        if (!found)
+        if (!coinControl.IsSelected(txin.prevout))
```

Comment:
Is the old and new code equivalent here? 
To me it looks like `coinControl.IsSelected` and the old code check for different things: IsSelected checks `coincontrol.setSelected`, whereas the old code just looks at `wtx` and `tx`.
Can you explain?